django/cantusdb_project/main_app/tests/test_views.py

```diff
@@ -4895,29 +4899,34 @@ def test_structure(self):
 
         first_chant = first_item["chant"]
         chant_keys = first_chant.keys()
-        expected_keys = [
+        expected_keys = {
```

Comment:
Also note that, while some APIs and code libraries do preserve order, [JSON Objects are not ordered](https://www.json.org/json-en.html).

> An object is an unordered set of name/value pairs. ...

So it's really in your best interests to not expect objects to be in any particular order, since you don't necessarily know if any particular library will preserve it.

Reply:
Ah! Thanks...

also, in that case, I'm not sure what this is testing that is different than `test_values` immediately below this function.

Reply:
I changed the ordering in the code so that it's clear how it corresponds to the specification. But since JSON Objects are unordered, I figured the test would be clearer if we worked with unordered sets - otherwise future developers might think that the ordering is important.

I see where you're coming from with your thought about `test_values`. When I initially implemented/recreated this API, it took me a while to wrap my head around all the levels of nesting in the final response. If I recall, this test was an effort to communicate how the request was meant to be structured (in a different way than the couple of other forms of documentation we also had). I guess it's not strictly necessary considering we also have `test_values`. If you think it impedes understanding on the whole, we can remove it - let me know what you think.